code/modules/atmospherics/machinery/air_alarm/_air_alarm.dm

```diff
@@ -137,6 +137,9 @@ GLOBAL_LIST_EMPTY_TYPED(air_alarms, /obj/machinery/airalarm)
 /obj/machinery/airalarm/Destroy()
 	if(my_area)
 		my_area = null
+	if(connected_sensor)
+		disconnect_sensor()
```

Comment:
I noticed the `disconnect_sensor` proc seems to run code after disconnect, namely registering another signal and setting area, which seems inappropriate in `Destroy`.

Reply:
True true. I included only the relevant parts of `disconnect_sensor` cleanup via ee87722